src/ghc.jl

```diff
@@ -0,0 +1,98 @@
+"""
+    GHC(k, variogram)
+
+Geostatistical Hierarchical Clustering.
+
+Long description....
+
+## References
+...
+"""
+struct GHC <: PartitionMethod
+  variogram::V
+  k::Int
+end
+
+function partition(data::Data, method::GHC,kernel_type,merge_criterion)
+  # find lambda
+  𝛌 = choose_𝛌(data,35) #literature rule of thumb
+  # initialize history to store cluster/S(q) data
+  n = length(data) #number of points
+  h = Array{partition_info, 1}(undef, n)
+  C = 1:n #most granular partiton
+  d = zeros(n,n)#trivial placeholder for d
+  for q in 1:n
+    # find cluster partition C
+    C = merge_clusters(data,C,d,"avg")
+    # compute S(q)
+    S = compute_silhouette(h,q,C)
+    # store info (C,S) in history
+    h[q]=(C,S)
+  end
+  # select clustering partition with best S(q)
+  best_C = h[argmax (x -> x.C,h)].C
+  return best_C
+end
+
+function ghc_kernel(𝛌,x,s,type)
+    #uniform
+    #triangular
+    #Epanechnikov
+    #Gaussian
```

Comment:
So these models are variogram models already available in Variography.jl right? Let's use them 👍🏽

Reply:
I'm not sure I see all of them there